### DISCUSSION TITLE
(~9/6 금) 불특정 사용자 배포 기능 논의

### DISCUSSION THREAD

**Discussion by jminkkk** [woowacourse-teams/2024-code-zap#578] [이야기해봐요]
이번주 금요일(6일) 예정된 배포가 있습니다.

해당 날짜까지 배포를 하기 위해 추가 기능 또는 개선해야 할 점 등등에 대해 논의해봅시다.

## 1. Swagger 접근 제한 (켬미)
외부인이 우리 서버의 API 명세를 확인할 수 없도록 한다.

### 방식
1. try it out 막는다. 
2. 비밀번호 입력으로 막는다. 

### 결론 
시간 안에 하기 위해 1번 하고 시간되면 2번

### 방식
사이트 푸터에 코드잽 이메일과 함께 문의 메시지를 등록

### 결론 
하자

## 4. 계정 찾기
회원이 본인의 아이디를 잊어버렸을 때 어떻게 찾아주어야 할까?

### 결론
찾아줄 방법이 없다 ㅠㅠ 

### 결론 
한다

### 에러 알림 (헤인)
400 대 에러 발생 시 사용자에게 에러 정보를 전달하지 않는 것들에 대해 처리한다.

### 길이 제한 (월하)
특정 길이 이상 입력 시 프론트가 입력 아예 불가능하도록 막기

### 로그인 (마위)
로그인을 했는데 ‘로그인 하러 가기’가 있음

**Comment by zangsu** [comment]
정말정말 있어야 하는 기능만 적어 보았어요. 
이미 리뷰를 받았고, 반영하면 좋지만 문제는 아닌 기능은 일단 제외했습니다.

- 문의 기능
  - 버그 리포트
  - 건의 사항
  - 문의 사항 
  - etc...
- 보안
  - 비밀번호 암호화
  - 문서화 제공되지 않도록 ~ 
- 계정
  - 계정 이름을 잃어버렸을 때 어떻게 찾아줄지 대책 마련

**Comment by kyum-q** [comment]
### 금요일 배포 시

#### 에러 해결

- 400 대 에러 시 프론트측에서도 처리 필요 (업로드 실패 시 마이페이지로 넘어감)
- 길이 제한에 맞게 입력 막기 (413 에러)
- 로그인을 했는데 ‘로그인 하러 가기’가 있음
- 등등등

#### 기능
- 비밀번호 암호화
- 문의하기

**Comment by jminkkk** [comment]
- 비밀번호 암호화
- swagger url 막기
- cors origin 변경
- 각종 문의사항 받을 연락처 페이지 (폼, 메일 등등)

+) ~~배포와 별개이지만,,, 금요일까지 각자 레벨3 문서화 해야하는 부분에 대해 문서화가 완료되었으면 좋겠어요~~

**Comment by Hain-tain** [comment]
- 프론트는 지금 크게 바꿀 수 있는 것 없기 때문에 크게 기능 추가할 건 없을 것 같습니다.
- 대신 사소한 변경들만 해도 좋을 것 같아요.
  - '설명 작성해주세요' 대신 '이걸 언제 다시 쓸 수 있을까요?' 등 언어만 변경 
  - 로그인 후 랜딩페이지의 ‘로그인하러가기’ 안 보이게 변경 
  - 등등
